[Bug] wireshark build points to wrong pulsar-common directory

### Search before asking

- [X] I searched in the [issues](https://github.com/apache/pulsar-client-cpp/issues) and found nothing similar.


### Version

3.1.1. Reproduced on M1 macbook pro but it should affect all OS are affected since it's a bad directory link.

### Minimal reproduce step

1. On a M1 macbook, run steps on https://github.com/apache/pulsar-client-cpp to install dependencies.
2. Run `cmake` command to prepare wireshark build:
```cmake -DBUILD_WIRESHARK=ON -DGTEST_INCLUDE_PATH=/usr/local/Cellar/googletest/1.13.0/include/gtest -DGMOCK_INCLUDE_PATH=/usr/local/Cellar/googletest/1.13.0/include/gmock .```
3. make pulsar-dissector points to a bad directory for pulsar-common.
```make pulsar-dissector
make[3]: *** No rule to make target `/Users/lin/git/pulsar-common/src/main/proto/PulsarApi.proto', needed by `generated/lib/PulsarApi.pb.cc'.  Stop.
make[2]: *** [wireshark/CMakeFiles/pulsar-dissector.dir/all] Error 2
make[1]: *** [wireshark/CMakeFiles/pulsar-dissector.dir/rule] Error 2
make: *** [pulsar-dissector] Error 2```

### What did you expect to see?

`make pulsar-dissector` should be successful.

### What did you see instead?

Error message that points to a bad directory for `pulsar-common`.

### Anything else?

This should be caused by the migration from the core pulsar directory. The script need to be updated to point to the right `pulsar-common` source code for `PulsarApi.proto`.

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

--- a/wireshark/CMakeLists.txt
+++ b/wireshark/CMakeLists.txt
@@ -60,9 +60,9 @@ include_directories(${LIB_AUTOGEN_DIR})
 set(PROTO_SOURCES ${LIB_AUTOGEN_DIR}/PulsarApi.pb.cc ${LIB_AUTOGEN_DIR}/PulsarApi.pb.h)
 ADD_CUSTOM_COMMAND(
         OUTPUT ${PROTO_SOURCES}
-        COMMAND ${PROTOC_PATH} -I ../../pulsar-common/src/main/proto ../../pulsar-common/src/main/proto/PulsarApi.proto --cpp_out=${LIB_AUTOGEN_DIR}
+        COMMAND ${PROTOC_PATH} -I ../proto ../proto/PulsarApi.proto --cpp_out=${LIB_AUTOGEN_DIR}
         DEPENDS
-        ../../pulsar-common/src/main/proto/PulsarApi.proto
+        ../proto/PulsarApi.proto
         WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})
 link_libraries(${Protobuf_LIBRARIES})
 

--- a/wireshark/README.md
+++ b/wireshark/README.md
@@ -70,7 +70,6 @@ $ sudo apt install wireshark-dev
 Compile the dissector.
 
 ```shell
-cd pulsar-client-cpp
 cmake -DBUILD_WIRESHARK=ON .
 make pulsar-dissector
 ```
@@ -91,18 +90,18 @@ You can see the location of personal plugins, which is important for the next st
 
 Example
 
-Wireshark 3.6.0 on macOS
+Wireshark 4.0.3 on macOS
 
 ```shell
-~/.local/lib/wireshark/plugins/3-6/
+~/.local/lib/wireshark/plugins/4-0/
 ```
 
 ### Copy Wireshark dissector to appropriate location
 
 ```shell
-mkdir -p ~/.local/lib/wireshark/plugins/3-6/epan
-cd pulsar-client-cpp/wireshark
-cp pulsar-dissector.so ~/.local/lib/wireshark/plugins/3-6/epan
+mkdir -p ~/.local/lib/wireshark/plugins/4-0/epan
+cd wireshark
+cp pulsar-dissector.so ~/.local/lib/wireshark/plugins/4-0/epan
 ```
 
 ### Complete installation

--- a/wireshark/pulsarDissector.cc
+++ b/wireshark/pulsarDissector.cc
@@ -16,15 +16,13 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-#include <config.h>
 #include <epan/column-utils.h>
 #include <epan/dissectors/packet-tcp.h>
-#include <epan/expert.h>
 #include <epan/packet.h>
-#include <epan/prefs.h>
 #include <epan/proto.h>
 #include <epan/value_string.h>
 #include <glib.h>
+#include <ws_version.h>
 #include <wsutil/nstime.h>
 
 #include "PulsarApi.pb.h"
@@ -1055,6 +1053,14 @@ static int dissect_pulsar_message(tvbuff_t* tvb, packet_info* pinfo, proto_tree*
             break;
         case BaseCommand::TC_CLIENT_CONNECT_RESPONSE:
             break;
+        case BaseCommand::WATCH_TOPIC_LIST:
+            break;
+        case BaseCommand::WATCH_TOPIC_LIST_SUCCESS:
+            break;
+        case BaseCommand::WATCH_TOPIC_UPDATE:
+            break;
+        case BaseCommand::WATCH_TOPIC_LIST_CLOSE:
+            break;
     }
 
     return maxOffset;
@@ -1210,8 +1216,8 @@ void proto_register_pulsar() {
 extern "C" {
 
 extern __attribute__((unused)) WS_DLL_PUBLIC_DEF const gchar plugin_version[] = VERSION;
-extern __attribute__((unused)) WS_DLL_PUBLIC_DEF const int plugin_want_major = VERSION_MAJOR;
-extern __attribute__((unused)) WS_DLL_PUBLIC_DEF const int plugin_want_minor = VERSION_MINOR;
+extern __attribute__((unused)) WS_DLL_PUBLIC_DEF const int plugin_want_major = WIRESHARK_VERSION_MAJOR;
+extern __attribute__((unused)) WS_DLL_PUBLIC_DEF const int plugin_want_minor = WIRESHARK_VERSION_MINOR;
 
 WS_DLL_PUBLIC void plugin_register(void);
 

